src/main/kotlin/no/nav/syfo/Bootstrap.kt

```diff
@@ -260,94 +267,177 @@ suspend fun blockingApplicationLogic(
+) {
 
-            try {
-                val doctor = fetchDoctor(helsepersonellv1, receivedSykmelding.personNrLege)
-
-                val helsepersonellKategoriVerdi = finnAktivHelsepersonellAutorisasjons(doctor)
-
-                when {
-                    results.any { rule -> rule.status == Status.MANUAL_PROCESSING } ->
-                        produceManualTask(kafkaproducerCreateTask, receivedSykmelding, validationResult, navKontorNr, logKeys, logValues)
-                    else -> sendInfotrygdOppdatering(
-                            infotrygdOppdateringProducer,
-                            session,
-                            logKeys,
-                            logValues,
-                            receivedSykmelding.fellesformat,
-                            InfotrygdForespAndHealthInformation(infotrygdForespResponse, healthInformation),
-                            receivedSykmelding.personNrPasient,
-                            receivedSykmelding.sykmelding.signaturDato.toLocalDate(),
-                            helsepersonellKategoriVerdi,
-                            receivedSykmelding.tssid,
-                            navKontorNr)
-                }
-                val currentRequestLatency = requestLatency.observeDuration()
-
-                log.info("Message($logKeys) got outcome {}, {}, processing took {}s",
-                        *logValues,
-                        keyValue("status", validationResult.status),
-                        keyValue("ruleHits", validationResult.ruleHits.joinToString(", ", "(", ")") { it.ruleName }),
-                        keyValue("latency", currentRequestLatency))
-            } catch (e: IHPR2ServiceHentPersonMedPersonnummerGenericFaultFaultFaultMessage) {
-                val validationResultBehandler = ValidationResult(
-                        status = Status.MANUAL_PROCESSING,
-                        ruleHits = listOf(RuleInfo(
-                                ruleName = "BEHANDLER_NOT_IN_HPR",
-                                messageForSender = "Den som har skrevet sykmeldingen din har ikke autorisasjon til dette.",
-                                messageForUser = "Behandler er ikke register i HPR"))
-                )
-                RULE_HIT_STATUS_COUNTER.labels(validationResultBehandler.status.name).inc()
-                log.warn("Behandler er ikke register i HPR")
-                produceManualTask(kafkaproducerCreateTask, receivedSykmelding, validationResultBehandler, navKontorNr, logKeys, logValues)
-            }
+    try {
+        val doctor = fetchDoctor(helsepersonellv1, receivedSykmelding.personNrLege)
+
+        val helsepersonellKategoriVerdi = finnAktivHelsepersonellAutorisasjons(doctor)
+
+        when {
+            validationResult.status in arrayOf(Status.MANUAL_PROCESSING) ->
+                produceManualTask(kafkaproducerCreateTask, receivedSykmelding, validationResult, navKontorNr, loggingMeta)
+            else -> sendInfotrygdOppdatering(
```

Comment:
denne funksjonene kan vel bare ta inn sykmeldignen istedet for 4 separate felter?

Reply:
Ifix